spark/v3.1/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java

```diff
@@ -212,13 +213,7 @@ public void testDropIdentityPartition() {
 
     table.refresh();
 
-    PartitionSpec expected =
-        PartitionSpec.builderFor(table.schema())
-            .withSpecId(1)
-            .alwaysNull("category", "category")
-            .build();
-
-    Assert.assertEquals("Should have new spec field", expected, table.spec());
+    Assert.assertTrue("New spec must be unpartitioned", table.spec().isUnpartitioned());
```

Comment:
Do we want to keep a test for the V1 Behavior? We haven't actually deprecated it.

Reply:
These are being tested against both now.